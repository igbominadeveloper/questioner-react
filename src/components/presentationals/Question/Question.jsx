import React from 'react';

const Question = () => {
  return (
    <div className="question flex p-1 flex-column">
      <div className="question-body flex">
        <div className="avatar bg-black">
          <p>ID</p>
        </div>
        <div className="question-details">
          <div className="question-content">
            <div className="upper-details flex space-between">
              <p className="question-user">
                <a href="profile.html" className="text-black">
                  Igbominadeveloper
                </a>
              </p>
              <p className="text-light question-time">
                <i className="fa fa-clock-o" />2 months ago
              </p>
            </div>
            <p className="text-light mt-1">
              Lorem ipsum dolor sit amet consectetur, adipisicing elit.
              Assumenda laborum cumque praesentium impedit obcaecati est eaque
              consectetur nisi tempore cupiditate?
            </p>
          </div>
          <div className="question-stats flex space-between mt-1">
            <p className="comment text-light">
              <i className="fa fa-reply" />1
            </p>
            <p className="like text-light upvote">
              <i className="fa fa-thumbs-o-up" />
              <span>5</span>
            </p>
            <p className="dislike text-light">
              <i className="fa fa-thumbs-o-down" />
              <span>1</span>
            </p>
          </div>
        </div>
      </div>

      <div className="question-comments">
        <div className="single-comment flex p-1">
          <div className="avatar bg-black">
            <p>ID</p>
          </div>
          <div className="comment-details">
            <div className="comment-content">
              <div className="upper-details flex space-between">
                <p className="comment-user">
                  <a href="profile.html" className="text-black">
                    Igbominadeveloper
                  </a>
                </p>
                <p className="text-light comment-time">
                  <i className="fa fa-clock-o" />2 months ago
                </p>
              </div>
              <p className="text-light mt-1">
                Lorem ipsum dolor sit amet consectetur, adipisicing elit.
                Assumenda laborum cumque praesentium impedit obcaecati est eaque
                consectetur nisi tempore cupiditate?
              </p>
            </div>
          </div>
        </div>
      </div>
      <div className="new-comment">
        <form action="" id="comment-form" className="flex flex-column">
          <div className="form-group">
            <textarea
              className="form-control"
              name="comment"
              id="comment"
              placeholder="Your comment"
            />
          </div>
          <button className="btn-primary p-1 shadow">post</button>
        </form>
      </div>
    </div>
  );
};

export default Question;
