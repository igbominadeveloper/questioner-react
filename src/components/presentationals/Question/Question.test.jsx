import React from 'react';
import { shallow } from 'enzyme';

import Question from './Question';

const props = {
  question: {
    user: {
      firstname: '',
      lastname: '',
    },
  },
};

describe('Question', () => {
  describe('Render', () => {
    it('renders without error', () => {
      const component = shallow(<Question {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
