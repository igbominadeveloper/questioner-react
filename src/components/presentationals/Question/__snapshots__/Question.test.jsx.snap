// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Render renders without error 1`] = `
<div
  className="question flex p-1 flex-column"
>
  <div
    className="question-body flex"
  >
    <div
      className="avatar bg-black"
    >
      <p />
    </div>
    <div
      className="question-details"
    >
      <div
        className="question-content"
      >
        <div
          className="upper-details flex space-between text-bold"
        >
          <p
            className="question-user"
          >
            <Link
              className="text-black"
              to="#"
            >
              Â 
            </Link>
          </p>
          <p
            className="text-light question-time"
          >
            <i
              className="fa fa-clock-o"
            />
            a few seconds ago
          </p>
        </div>
        <p
          className="text-light mt-1"
        />
      </div>
      <div
        className="question-stats flex space-between mt-1"
      >
        <p
          className="comment text-light"
        >
          <i
            className="fa fa-reply"
          />
          0
        </p>
        <p
          className="like text-light upvote"
          onClick={[Function]}
        >
          <i
            className="fa fa-thumbs-o-up"
          />
          <span />
        </p>
        <p
          className="dislike text-light"
          onClick={[Function]}
        >
          <i
            className="fa fa-thumbs-o-down"
          />
          <span />
        </p>
      </div>
    </div>
  </div>
  <div
    className="question-comments"
  >
    <div
      className="single-comment flex p-1"
    >
      <div
        className="avatar bg-black"
      >
        <p>
          ID
        </p>
      </div>
      <div
        className="comment-details"
      >
        <div
          className="comment-content"
        >
          <div
            className="upper-details flex space-between"
          >
            <p
              className="comment-user"
            >
              <a
                className="text-black"
                href="profile.html"
              >
                Igbominadeveloper
              </a>
            </p>
            <p
              className="text-light comment-time"
            >
              <i
                className="fa fa-clock-o"
              />
              2 months ago
            </p>
          </div>
          <p
            className="text-light mt-1"
          >
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Assumenda laborum cumque praesentium impedit obcaecati est eaque consectetur nisi tempore cupiditate?
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="new-comment"
  >
    <form
      action=""
      className="flex flex-column"
      id="comment-form"
    >
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          id="comment"
          name="comment"
          placeholder="Your comment"
        />
      </div>
      <button
        className="btn-primary p-1 shadow"
      >
        post
      </button>
    </form>
  </div>
</div>
`;
